06_puppy/mohitnag/store_test.go

```diff
@@ -0,0 +1,126 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type storer struct {
+	suite.Suite
+	store      Storer
+	storerType func() Storer
+}
+
+func (s *storer) SetupTest() {
+	s.store = s.storerType()
+}
+
+func (s *storer) TestCreatePuppy() {
+	assert := assert.New(s.T())
+	puppy := Puppy{ID: 1, Breed: "dog", Colour: "white", Value: "$2"}
+	testCases := []struct {
+		Scenario      string
+		Input         Puppy
+		ExpectedError string
+	}{
+		{"Create Puppy", puppy, "<nil>"},
+		{"Creating already existing Puppy should fail", puppy, "puppy with Id 1 already exists"},
+	}
```

Comment:
I like this way of setting up the tests where there is one test function for each function being tested with multiple 'Scenario's setup.  I have separate test functions for success and fail... hopefully the other reviewers will like to too...

Reply:
👍